WcaOnRails/spec/models/user_spec.rb

```diff
@@ -80,6 +80,13 @@
     expect(delegate).to be_valid
   end
 
+  it "requires senior delegate if delegate status allows it" do
+    delegate = FactoryBot.create :delegate
+
+    delegate.senior_delegate = nil
+    expect(delegate).to be_invalid_with_errors(senior_delegate: ["must be present"])
+  end
```

Comment:
You can specify attributes during creation to simplify this:
```rb
it "requires senior delegate if delegate status allows it" do
  delegate = FactoryBot.build :delegate, senior_delegate: nil
  expect(delegate).to be_invalid_with_errors(senior_delegate: ["must be present"])
end
```
Also note that I used `build` instead` of `create`. It's fine both ways, but given we know that creation should fail, there's no reason to try saving it to the database.

Reply:
Noted, thanks!